utils/filesystem/files_test.go

```diff
@@ -1306,6 +1306,37 @@ func TestFilepathStem(t *testing.T) {
 	})
 }
 
+func TestUnzipFileCountLimit(t *testing.T) {
+	fs := NewFs(StandardFS)
+
+	testInDir := "testdata"
+	limits := NewLimits(1<<30, 10<<30, 10)
+
+	// above file count limit test
+	testFile := "abovefilecountlimitzip"
+	srcPath := filepath.Join(testInDir, testFile+".zip")
+
+	destPath, err := fs.TempDirInTempDir("unzip-limits-")
+	assert.NoError(t, err)
+
+	defer fs.Rm(destPath)
+
+	_, err = fs.UnzipWithContextAndLimits(context.TODO(), srcPath, destPath, limits)
+	assert.Error(t, err)
+
+	//below file count limit test
+	testFile = "abovefilecountlimitzip"
+	srcPath = filepath.Join(testInDir, testFile+".zip")
+
+	destPath, err = fs.TempDirInTempDir("unzip-limits-")
+	assert.NoError(t, err)
+
+	defer fs.Rm(destPath)
+
+	_, err = fs.UnzipWithContextAndLimits(context.TODO(), srcPath, destPath, limits)
+	assert.NoError(t, err)
```

Comment:
If the file is above the limit then an error should be raised
```suggestion
	assert.Error(t, err)
```
you should also decide on the type of error returned by doing a task like

The error returned should be of the type of one of the errors in this list https://github.com/ARM-software/golang-utils/blob/master/utils/commonerrors/errors.go

```
assert.True(t, commonerrors.Any(err, commonerrrors.ErrTooLarge) 
```